src/pages/oversikt/filtrering/Filtrering.tsx

```diff
@@ -48,12 +55,32 @@ const Filtrering: FunctionComponent<RouteComponentProps> = props => {
             );
         }
 
-        const queryParams = lagQueryParams(filterFraUrl);
-        props.history.replace({ search: queryParams });
+        setFilterIUrl(filterFraUrl);
+    };
+
+    const brukMellomlagretFilter = (mellomlagretFilter: Filter) => {
+        setValgteKriterier(mellomlagretFilter);
+        setFilterIUrl(mellomlagretFilter);
+    };
+
+    const brukFilterFraUrl = (filterFraUrl: Filter) => {
+        setValgteKriterier(filterFraUrl);
+        setMellomlagretFilter(filterFraUrl);
+    };
+
+    const setFilterIUrl = (filter?: Filter) => {
```

Comment:
Kunne man hatt en one-liner for denne funksjonen?
Typ `props.history.replace({ search: lagQueryParams(filter) || '' });`
Eller kanskje det blir mindre leselig da...